lint/parenthesis-checker.go

```diff
@@ -0,0 +1,46 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+// ParenthesisChecker ...
+type ParenthesisChecker struct {
+	ctx *Context
+
+	warnings []Warning
+}
+
+func NewParenthesisChecker(ctx *Context) *ParenthesisChecker {
+	return &ParenthesisChecker{
+		ctx: ctx,
+	}
+}
+
+func (c *ParenthesisChecker) Check(f *ast.File) []Warning {
+	c.warnings = c.warnings[:0]
+	for _, decl := range collectFuncDecls(f) {
+		if decl.Type.Results == nil {
+			continue
+		}
+		for _, res := range decl.Type.Results.List {
+			c.validateResultDecl(res)
+		}
+	}
+	return c.warnings
+}
+
+func (c *ParenthesisChecker) validateResultDecl(f *ast.Field) {
```

Comment:
We could work on this method design later to make it possible to re-use it for other AST nodes check (we don't want to be limited to results).

But for now, let's change `validate` to `check` for consistency (`checkResultDecl`).